### PR TITLE
🔎 Added meta descriptions to all pages using gpt 4

### LINKED ISSUE #2735: 🔎 Semrush - Weekly track and improve SEO #8
Improve SEO using semrush

URL: [https://www.semrush.com/siteaudit/campaign/16078651/review#overview](https://aus01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fwww.semrush.com%2Fsiteaudit%2Fcampaign%2F16078651%2Freview%23overview&data=05%7C02%7CJeoffreyFischer%40ssw.com.au%7C844a6b7d94314a096dbd08dc59c68ac7%7Cac2f7c34b93548e9abdc11e5d4fcb2b0%7C0%7C0%7C638483956578243635%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C0%7C%7C%7C&sdata=cYelQwC0XmTPzXZSING%2F09XjZnN9PVEBAcxSwyTvO5c%3D&reserved=0)

### PR DESCRIPTION
**Pull request by Calinator444**
CC: @camillars, @adamcogan, @bradystroud 

### Description 

- Fixed #2735 

**Tldr**: I've run a Python script to generate meta descriptions for all of the pages on our main website using the content of the page (All of which are under 150 characters).

**Long version**:
Robots are definitely coming for my job, but that doesn't mean I can't use AI to help us get some of that SEO cheddar for the time being.
BiggiecheeseRatGIF

 I've run a Python script that uses the file structure of the repo to determine which pages should exist on the live site, scrapes the content of the page, and sends it to chatGPT using an API so that it could summarize the content into meta descriptions. I couldn't pass the content of each Markdown file directly into ChatGPT because they're broken up into multiple files. Instead I've used a python library called Selenium to load each page into the browser, wait for the client side scripts to finish running, and then create a snapshot of the page. Afterwards I used another Python library to strip all of the HTML tags from the page leaving only the written content. Finally I sent the written content to chat GPT with a system prompt telling it to summarize the content into a meta description under 150 characters. I had to decrease the temperature of the response to get this to work because when chatGPT's responses are too open-ended and creative it surpassed the character limit. Finally I added a check in the script to query the API again if the response was over 150 characters. Meaning all of these descriptions should definitely be under  150 characters.

### List of Pages
I've included the list of page URLs affected along with their corresponding meta tags to make this easier to review. The page list was also generated by the script so there shouldn't be anything missing.
[seo_descriptions.xlsx](https://github.com/user-attachments/files/15835951/seo_descriptions.xlsx)

### Affected routes
https://ssw.com.au/company/clients
https://ssw.com.au/company/resources
https://ssw.com.au/company/clients/auctions-plus
https://ssw.com.au/company/clients/aurecon
https://ssw.com.au/company/clients/ausgrid
https://ssw.com.au/company/clients/bce
https://ssw.com.au/company/clients/bupa
https://ssw.com.au/company/clients/emove
https://ssw.com.au/company/clients/estate-master
https://ssw.com.au/company/clients/event-cinemas
https://ssw.com.au/company/clients/hearing-australia
https://ssw.com.au/company/clients/hutchison-weller
https://ssw.com.au/company/clients/maximus-imprint
https://ssw.com.au/company/clients/medibank
https://ssw.com.au/company/clients/microsoft-sharepoint-office-forum
https://ssw.com.au/company/clients/pwc
https://ssw.com.au/company/clients/shepherd-centre
https://ssw.com.au/company/clients/sydney-uni
https://ssw.com.au/company/clients/tafe
https://ssw.com.au/company/clients/toll
https://ssw.com.au/company/clients/vulcan-steel
https://ssw.com.au/company/clients/worley-parsons
https://ssw.com.au/company/index/
https://ssw.com.au/employment/
https://ssw.com.au/live/
https://ssw.com.au/logo/ai-hackday
https://ssw.com.au/logo/ai-workshop
https://ssw.com.au/logo/angular-hackday
https://ssw.com.au/logo/angular-workshop
https://ssw.com.au/logo/azure-workshop
https://ssw.com.au/logo/ca-workshop
https://ssw.com.au/logo/codeauditor
https://ssw.com.au/logo/dory
https://ssw.com.au/logo/eagleeye
https://ssw.com.au/logo/firebootcamp
https://ssw.com.au/logo/linkauditor
https://ssw.com.au/logo/maui-hackday
https://ssw.com.au/logo/netug
https://ssw.com.au/logo/past
https://ssw.com.au/logo/smashingbarrier
https://ssw.com.au/logo/sophiebot
https://ssw.com.au/logo/sophiehub
https://ssw.com.au/logo/ssw-rewards
https://ssw.com.au/logo/ssw-rules
https://ssw.com.au/logo/ssw-tv
https://ssw.com.au/logo/ssw
https://ssw.com.au/logo/sugarlearning
https://ssw.com.au/logo/timepro
https://ssw.com.au/logo/yakshaver
https://ssw.com.au/netug/
https://ssw.com.au/netug/present
https://ssw.com.au/advertising
https://ssw.com.au/booknow
https://ssw.com.au/domains-for-sale
https://ssw.com.au/hackday
https://ssw.com.au/new-zealand
https://ssw.com.au/products/rewards

### CODE REVIEW

File: content/seo_description_filler.py
Comment (bradystroud) [inline content/seo_description_filler.py:1]: Move this to a scripts folder, also add a README to the folder that explains what this script was used for (for future reference)

e.g. I wrote this for the rules one 
https://github.com/SSWConsulting/SSW.Rules.Content/blob/main/scripts/generateSeoDescriptions/README.md

**Review by bradystroud — CHANGES_REQUESTED** [review]
Hey Caleb, the descriptions look good! Only feedback is I think a bit of documentation would be really good here

File: content/seo_description_filler.py
Comment (Calinator444) [reply content/seo_description_filler.py:1]: Done - I added some docs for running the script. If it will need to be routinely run switching away from the ChatGPT implementation should be high priority.

**Review by bradystroud — APPROVED** [review]
LGTM
